### PR TITLE
Added `bytes32` as target type for `decimal` conversion

### PR DESCRIPTION
**Pull request by jakerockland**
### What I did

Added support for conversion _to_ `bytes32` _from_ `decimal`. Minor part of #1093

### How I did it

Minor change to `convert.py`.

### How to verify it

Run the tests:

`make test`

Or more specifically:

`pytest -k test_convert_to_bytes32`

### Description for the changelog

Added support for conversion _to_ `bytes32` _from_ `decimal`.

### Cute Animal Picture

image

### CODE REVIEW

File: tests/parser/functions/test_convert_to_bytes32.py [hunk 1]
```diff
@@ -80,3 +80,64 @@ def testConvertBytes32(flag: bool) -> bytes32:
...
+    foobarVal = c.foobar()
+    assert fooVal == (b"\x00" * 32)
+    assert len(fooVal) == 32
+    assert foobarVal == (b"\x00" * 32)
+    assert len(foobarVal) == 32
+    assert fooVal == foobarVal
+
+    hooVal = c.hoo()
+    hoonarVal = c.hoonar()
+    assert hooVal[0:11] == (b"\xff" * 11)
```
Comment (fubuloubu) [inline tests/parser/functions/test_convert_to_bytes32.py]: Definitely need some comments here and similar other lines in this test to describe why certain bit patters are set.

Will docs be added to describe what the conversion will look like?

Comment (jakerockland) [reply tests/parser/functions/test_convert_to_bytes32.py]: @fubuloubu great point, I'll add some comments noting the decimal precision and why these tests are like this + also will add more tests

I'll also add to the VIP at #1093 and to PR #1501 to describe in a bit more detail what this specific conversion looks like, as the decimal precision part is a bit unintuitive 😅 

Will take care of this tonight 👍

Comment (fubuloubu) [reply tests/parser/functions/test_convert_to_bytes32.py]: You the man!

Comment (jakerockland) [reply tests/parser/functions/test_convert_to_bytes32.py]: @fubuloubu updated! 😄

**Comment by jakerockland** [comment]
Just updated the VIP #1093 and pending docs PR #1501 with additional clarification as well @fubuloubu 😄

File: tests/parser/functions/test_convert_to_bytes32.py [hunk 2]
```diff
@@ -80,3 +80,107 @@ def testConvertBytes32(flag: bool) -> bytes32:
...
+    foobarVal = c.foobar()
+    assert fooVal == (b"\x00" * 32)
+    assert len(fooVal) == 32
+    assert foobarVal == (b"\x00" * 32)
+    assert len(foobarVal) == 32
+    assert fooVal == foobarVal
+
+    hooVal = c.hoo()
+    hoonarVal = c.hoonar()
+    _hoo = ((-2**127) * decimal_divisor).to_bytes(32, byteorder="big", signed=True)
```
Comment (davesque) [inline tests/parser/functions/test_convert_to_bytes32.py]: Maybe we should factor out these repeated `.to_bytes(...)` operations into a helper function at the top of this testing module.  Also, you can import `DECIMAL_DIVISOR` from `vyper.utils`.

**Comment by davesque** [comment]
Thanks for the work, @jakerockland !

File: tests/parser/functions/test_convert_to_bytes32.py [hunk 3]
```diff
@@ -80,3 +80,107 @@ def testConvertBytes32(flag: bool) -> bytes32:
...
+    return convert(self.kar, bytes32)
+    """
+
+    c = get_contract_with_gas_estimation(code)
+    decimal_divisor = 10000000000
+
+    fooVal = c.foo()
+    foobarVal = c.foobar()
+    assert fooVal == (b"\x00" * 32)
+    assert len(fooVal) == 32
```
[the remark this answers is not in the document]
Comment (jakerockland) [reply tests/parser/functions/test_convert_to_bytes32.py]: @davesque good point, these is a bit of left-over bad testing from the original PR before I updated the tests—I'll clean this up 👍

File: tests/parser/functions/test_convert_to_bytes32.py [hunk 4]
```diff
@@ -80,3 +80,107 @@ def testConvertBytes32(flag: bool) -> bytes32:
...
+def test_convert_from_decimal(get_contract_with_gas_estimation):
+    code = """
+bar: decimal
+nar: decimal
+mar: decimal
+jar: decimal
+kar: decimal
+
+@public
+def foo() -> bytes32:
```
[the remark this answers is not in the document]
Comment (jakerockland) [reply tests/parser/functions/test_convert_to_bytes32.py]: Sure thing @davesque will take care of this and the suggested refactoring below this evening.

**Comment by jakerockland** [comment]
@davesque just pushed a commit that incorporates all of your suggested changes above, may be worth making a code quality ticket to address auditing test methods across the testing suite and renaming needed functions to be more meaningful/readable, as the `foobar` thing is definitely present in other tests 😅I've written some of those `foo`s but _definitely_ not all of them 😛

**Comment by davesque** [comment]
@jakerockland Nice, thanks for the heads up.  I wasn't aware of that.

**Comment by jakerockland** [comment]
@davesque sure thing! Definitely would be a nice improvement in code-quality, but catching it new additions to the codebase like here is a good first step 😅

**Pull request merged**